tests/utils.py

```diff
@@ -0,0 +1,22 @@
+from pathlib import Path
+from hashlib import sha256
+from sha3 import keccak_256
+
+from ecdsa.curves import SECP256k1
+from ecdsa.keys import VerifyingKey
+from ecdsa.util import sigdecode_der
+
+
+ROOT_SCREENSHOT_PATH = Path(__file__).parent.resolve()
+
+# Check if a signature of a given message is valid
+def check_signature_validity(public_key: bytes, signature: bytes, message: bytes) -> bool:
```

Comment:
This is unused I think?
could it be used instead of hard coding expected signature?

Reply:
yes you're right. I will remove it.
I tried to use it to avoid hardcoded expected signature, but sadly the signature computed with this function does not match with the signature computed by the nano devices.

Reply:
This function doesn't compute the signature but check for validity againt the public key.
This is weird that it doesn't work.
Did you tried it with the right parameters? encoded the right way maybe? parameters should be equivalent to what is passed to the sign function in the app. Also maybe the app is using a different hash function?

Reply:
@xchapron-ledger Yes I understand. But it seems that signature sent by the device is truncated and the `verify `function called in `check_signature_validity` returns an error (0x30 is expected instead of 0x10). Perhaps the parameters of `verify` function are not the right ones. If you have any clue on this, it will be very useful. Here I'm not talking about the parameters of `check_signature_validity`.

Reply:
@tdejoigny-ledger @xchapron-ledger It might be the reason that vechain uses `blake2b256` as the hash function.